### PR TITLE
Add StackBlitz examples for TypeScript

### PR DESCRIPTION
**Pull request by SamVerschueren**
I improved the TypeScript support on StackBlitz and added two TypeScript examples.

- [Basic example for AVA with TypeScript](https://stackblitz.com/github/SamVerschueren/ava/tree/stackblitz/examples/typescript/examples/typescript-basic?file=source%2Ftest.ts&terminal=test&view=editor)
- [TypeScript example for typing `t.context`](https://stackblitz.com/github/SamVerschueren/ava/tree/stackblitz/examples/typescript/examples/typescript-context?file=source%2Ftest.ts&terminal=test&view=editor)

### CODE REVIEW

**Comment by SamVerschueren** [comment]
@sindresorhus I use a `tsconfig.json` file which extends from `@sindresorhus/tsconfig` for the examples. But `@sindresorhus/tsconfig` does not exist after the CI runs `npm install`. So we end up with this error.

image

Can I somehow disable the `tsconfig` inference and use the built-in `parserOptions.project`? I could disable linting of those files, but it's a bit of a bummer 🤔 .

File: examples/typescript-context/package.json [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+		"ts-node": "^10.0.0",
+		"typescript": "^4.3.4"
+	},
+	"ava": {
+		"extensions": [
+			"ts"
+		],
+		"require": [
+			"ts-node/register"
+		]
```
Comment (novemberborn) [inline examples/typescript-context/package.json:19]: Can we make this work with `@ava/typescript`? Perhaps the built-in compiler mode that's in a pre-release?

Comment (SamVerschueren) [reply examples/typescript-context/package.json]: Done! I totally forgot about `@ava/typescript`, my bad! This is much better. Also the `compile` flag seems to work with version 2 :).

File: .xo-config.json [hunk 2]
```diff
@@ -3,7 +3,8 @@
 		"media/**",
 		"test/config/fixtures/config-errors/test.js",
 		"test-tap/fixture/snapshots/test-sourcemaps/build/**",
-		"test-tap/fixture/report/edgecases/ast-syntax-error.cjs"
+		"test-tap/fixture/report/edgecases/ast-syntax-error.cjs",
+		"examples/typescript-*/**/*.ts"
```
Comment (SamVerschueren) [inline .xo-config.json:7]: Disabling it for all `typescript` projects now. It's the easiest solution I guess.

**Comment by SamVerschueren** [comment]
Not sure why, but sometimes some tests are flaky. Is that normal on Windows?

**Comment by novemberborn** [comment]
> Not sure why, but sometimes some tests are flaky. Is that normal on Windows?

Yes 🤷

**Pull request merged**